data/text_sampling/text_sampling.py

```diff
@@ -0,0 +1,919 @@
+import glob
+import math
+import os.path
+import random
+import re
+import string
+from functools import partial
+from typing import Callable, List
+
+import pandas as pd
+import yaml
+from utils import load_yaml, str_presenter
+
+DEFAULT_SIGNIFICANT_DIGITS = 3
+
+standard_tabular_text_templates = [
+    "The molecule with the {SMILES__description} representation of {SMILES#} has a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "The molecule {SMILES#} has a {TARGET__names__noun} of {TARGET#}.",
+    "Based on the {SMILES__description} representation {SMILES#}, the molecule has a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "Based on the representation {SMILES#}, the molecule has a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "The {SMILES__description} {SMILES#} represents a molecule that has a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "The {SMILES__description} {SMILES#} represents a molecule with a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "{SMILES#} represents a molecule that has a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "{SMILES#} represents a molecule with a {TARGET__names__noun} of {TARGET#}.",  # noqa: E501
+    "The {SMILES__description} {SMILES#} has a {TARGET__names__noun} of {TARGET#}.",
+    "The molecule {SMILES#} has a {TARGET__names__noun} of {TARGET#}.",
+    "{SMILES#} has a {TARGET__names__noun} of {TARGET#}.",
+    "The {TARGET__names__noun} of the {SMILES__description} {SMILES#} is:<EOI> {TARGET#}",  # noqa: E501
+    "The {TARGET__names__noun} of the {SMILES__description} {SMILES#} is<EOI> {TARGET#}.",  # noqa: E501
+]
+
+
+exclude_from_standard_tabular_text_templates = [
+    "ames_mutagenicity",  # because it is boolean target data
+    "bioavailability_ma_et_al",  # because it is boolean target data
+    "blood_brain_barrier_martins_et_al",  # because it is boolean target data
+    "chembl_v29",  # text only, no SMILES
+    "moses",  # SMILES only, has no target
+    "zinc",  # SMILES only, has no target
+]
```

Comment:
should we put this info into the `meta.yaml` files and not maintain it in two places?

Reply:
I added this to the to dos above!